src/main/java/com/google/sps/servlets/VerifyRoomServlet.java

```diff
@@ -30,23 +30,40 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that takes in the user's room Id and prints whether or not it exists
+/** 
+  Servlet that takes in the user's room Id and prints whether or not it exists
+  If so, it also checks if the user that is trying to join is on the Room's member list
 */
 @WebServlet("/verify-room")
 public final class VerifyRoomServlet extends HttpServlet {
-
+  private final String USER_PARAMETER =  "userEmail";
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
     response.setContentType("application/json");
-
+    String user = request.getParameter(USER_PARAMETER);
     String tempStringOfRoomId = request.getParameter(ServletUtil.INPUTTED_ID_TAG);
     long currentRoomId = Long.parseLong(tempStringOfRoomId);
     Room currentRoom = Room.fromRoomId(currentRoomId);
     
-    if(currentRoom == null){
+    if(currentRoom == null || !isUserOnMemberList(user, currentRoom)){
       response.getWriter().println(false);
-    } else {
-      response.getWriter().println(true);
+      return;
```

Comment:
Simply change to: 

response.getWriter().println(currentRoom == null || !isUserOnMemberList(user, currentRoom);

and you can get rid of if-else branch.